libs/sr-ui/src/Hook/useEquippedRelics.ts

```diff
@@ -1,54 +1,40 @@
-import type { CharacterKey, RelicSlotKey } from '@genshin-optimizer/sr-consts'
-import type { ICachedRelic } from '@genshin-optimizer/sr-db'
-import { useEffect, useState } from 'react'
+import {
+  allRelicSlotKeys,
+  type RelicSlotKey,
+} from '@genshin-optimizer/sr-consts'
+import type {
+  ICachedRelic,
+  ICachedSroCharacter,
+} from '@genshin-optimizer/sr-db'
+import { objKeyMap, objMap } from '@genshin-optimizer/util'
+import { useEffect, useMemo, useState } from 'react'
 import { useDatabaseContext } from '../Context'
-import { useCharacter } from './useCharacter'
 
-export function useEquippedRelics(characterKey: CharacterKey | '') {
```

Comment:
Since most of the time you are getting `character` outside of this hook already in the first place, you are actually storing `character` twice. Since this hook is intended to handle `equippedRelics`,concern is shifted upwards.